docs/usage/MTCAPIDoc.md

```diff
@@ -0,0 +1,417 @@
+```
+
+- spec.analyzeImageCount: A boolean flag to enable analysis of image count. This is a required field.
+
+- spec.analyzeK8SResources: A boolean flag to enable analysis of k8s resources. This is a required field.
+
+- spec.analyzePVCapacity: A boolean flag to enable analysis of persistent volume capacity. This is a required field.
+
+- spec.listImages: A boolean flag to enable used in analysis of image count.
+
+- spec.listImagesLimit: An integer representing limit on image counts.
+
+- spec.migPlanRef: ObjectReference contains enough information to let you inspect or modify the referred object. This is a required field.
+
+## Mig Cluster
+
+Local Mig cluster: 
+```
+ 
+apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-local
+  namespace: openshift-migration
+spec:
+  # [!] Change isHostCluster to 'false' if you want to use a clusterRef and serviceAccountSecretRef
+  #     instead of using the mig-controller detected kubeconfig. Refer to mig-cluster-aws.yaml for an example.
+  isHostCluster: true
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+
+  Remote Mig cluster:
+
+  ```
+  apiVersion: migration.openshift.io/v1alpha1
+kind: MigCluster
+metadata:
+  labels:
+    controller-tools.k8s.io: "1.0"
+  name: migcluster-remote
+  namespace: openshift-config
+spec:
+  # [!] Change isHostCluster to 'true' if you'll be running the controller on this cluster.
+  #     Setting 'isHostCluster' to true will bypass using the clusterRef and serviceAccountSecretRef below.
+  isHostCluster: false
+
+  url: "https://my-remote-cluster-ip.nip.io:8443"
+
+  serviceAccountSecretRef:
+    name: sa-token-remote
+    namespace: openshift-config
+
+  # [!] Change refresh to 'true' to force a manual reconcile
+  refresh: false
+  ```
+- spec.azureResourceGroup: For azure clusters -- it's the resource group that in-cluster volumes use.
+
+- spec.caBundle: If the migcluster needs SSL verification for connections a user can supply a custom CA bundle. This field is required only when spec.Insecure is set false.
+  
+- spec.exposedRegistryPath: Stores the path of registry route when using direct migration.
+
+- spec.insecure: If set false, user will need to provide CA bundle for TLS connection to the remote cluster.
+
+- spec.isHostCluster: Specifies if the cluster is host (where the controller is installed) or not. This is a required field.
+
+- spec.refresh: If set True, forces the controller to run a full suite of validations on migcluster.
+
+- spec.restartRestic: An override setting to tell the controller that the source cluster restic needs to be restarted after stage pod creation.
+
+- spec.serviceAccountSecretRef: ObjectReference contains enough information to let you inspect or modify the referred object.
+
+- spec.url: Stores the url of the remote cluster. The field is only required for the source cluster object.
```

Comment:
spec.url: Is the "remote" cluster the target cluster?

Reply:
@apinnick "Remote" cluster is where the controller is not installed. It could be "target" or "source" depending upon where the mig-controller is intalled.